### ISSUE TITLE
Node.exe fails to terminate and pauses GHCJS-Boot until terminated manually.

### ISSUE THREAD

**Issue by BethAr** [ghcjs/ghcjs#341]
I managed to build GHCJS in Windows 7 successfully, but `GHCJS-Boot --dev` didn't finish by itself, being hung in a suspended state without any activity whatsoever. I left it hanging for over an hour while I did other work.

I noticed that `node.exe` was still running and decided to terminate it, and as soon as I did `GHCJS-Boot --dev` completed the installation by itself. I tested some basic compiling with GHCJS and everything seems to be working normally.

I did the whole installation process from scratch in a different computer, also with Windows 7, and the result was the same, `GHCJS-Boot --dev` wouldn't complete until I terminated `node.exe` manually.

I think the line it was hung was `Installed GHCJS-Base.`, but I might be paraphrasing.

**Comment by luite** [comment]
I've seen this problem intermittently too, but haven't had time to completely figure out what's the problem yet. I think it's one of the setup scripts hanging, but i don't know on what. I think this still happens on Windows 10 developer preview.

(Cabal compiles `Setup.hs` and then runs it, when building with GHCJS, it compiles the file to JS and then runs the program with node.js)

**Comment by BethAr** [comment]
The problem seems to be inconsequential. I'm yet to experience an unexpected behaviour when compiling. I would suggest adding an note to the installation guide, such as:

In case `GHCJS-Boot` stops responding with the `Installed GHCJS-Base` line (I will see if I can verify the line), it can usually be resumed by terminating the `node` process manually.

**Comment by BethAr** [comment]
The line it hangs is:
```
Installing library in
C:\Users\Beta\AppData\Roaming\ghcjs\i386-mingw32-0.1.0-7.10.1\ghcjs/lib\ghcjs-base-0.1.0.0
Registering ghcjs-base-0.1.0.0...
Installed ghcjs-base-0.1.0.0
```

After I termiate `node.exe` it produces a gigantic and ugly path concatenation and starts the following section:
```
===========================================================================

   Boot libraries installation for GHCJS

===========================================================================
 ```
Then it finishes succesfully.

**Comment by luite** [comment]
Ok, I can reproduce this now, Windows 8.1 32 bit

**Comment by luite** [comment]
And it's the template haskell runner that's somehow not terminated correctly

**Comment by luite** [comment]
Hmm, this thing is hard to reproduce, just had a boot finish succesfully

**Comment by BethAr** [comment]
All the machines I installed are of the same build, that is, they have the exact same specification, including OS and hardware, perhaps that is why they all pause at the same time? Either way I have not seen a single installation succeed without having to terminate node manually.

The only thing in common I can think of is that I updated the version of node on all machines before installing GHCJS, because they all had some old odd version installed. Perhaps there is something lingering data that is hampering node's termination?

**Comment by luite** [comment]
I still see this from time to time. Does anyone have a way to make the installation process hang by installing a separate cabal package, rather than running the booter? that'd make investigating much easier.

**Comment by BethAr** [comment]
@luite, i've never seen this except when running the booter. Since it is node.exe that hangs, I don't know how could I reproduce the problem by installing regular packages. Do you have any suggestion of a package that could provoke the same issue?

**Comment by tejon** [comment]
Seeing the same thing here -- just to add one small data point in case it helps narrow the search: `node.exe` is constantly allocating memory at about 4k/sec, but not registering on the CPU meter.

**Comment by luite** [comment]
fixed in `improved-base` branch with commit d0d1bcf263939792d0b073b9802bdb9511661bad
